pagopa/io-app-design-system
# feat: [IOBP-236] ListItemSwitch extension with loading state and badge

## Short description
This PR adds three more props to `ListItemSwitch` component:
- `isLoading`: while true it shows an activity indicator
- `badgeText`: A text to show inside a badge instead of the switch. If it is not null, you have to provide also a `badgeVariant` 
- `badgeVariant`: It indicates the badge variant to show;

## List of changes proposed in this pull request
- Added three new props, the `badgeText` and `badgeVariant` are conditional properties;
- Changed the position of the right component and aligned to the title;
- Added more examples about `ListItemSwitch` inside example app;

## How to test
Open the example app, navigate through `List items` section and check the new examples:

## Preview
https://github.com/pagopa/io-app-design-system/assets/34343582/bd154d69-b7af-44ac-b09d-34674c80ab9e
